src/localkiwixserver.cpp

```diff
@@ -65,8 +78,19 @@ void LocalKiwixServer::openInBrowser()
 void LocalKiwixServer::runOrStopServer()
 {
     if (!m_active) {
+        auto settingsManager = KiwixApp::instance()->getSettingsManager();
+        m_port = ui->PortChooser->text().toInt();
+        if (m_port == 0) {
+            m_port = settingsManager->getKiwixServerPort();
+        }
```

Comment:
Is there a situation where `ui->PortChooser->text().toInt()` returns 0 ?

Reply:
Yep, when the line edit is left empty

Reply:
Shouldn't be checked at UI level ?
And if it appears if the line is empty, it is better to check if the line is empty.

Reply:
:+1: Will do